### PR TITLE
Guard debug infra

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90128
* __->__ #90123
* #90124

Test it

cc @mlazos @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/guards.py [hunk 1]
```diff
@@ -869,6 +875,14 @@ def guard_fail_hook(
         elif isinstance(fail_reason, bool) and not fail_reason:
             reasons.append(part)
             break
+    try:
+        breakpoint()
```
Comment (ezyang) [inline torch/_dynamo/guards.py:879]: is this ready for review?

**Comment by ezyang** [comment]
needs more pr description

File: torch/_dynamo/guards.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/guards.py:879]: Yes, just one leftover

**Comment by voznesenskym** [comment]
Abandoned for https://github.com/pytorch/pytorch/pull/90130

**Pull request closed without merging**